src/utils/time.ts

```diff
@@ -35,6 +35,32 @@ export function formatDateFromBatchId(batchId: number): string {
   return formatDistanceToNow(date, { addSuffix: true })
 }
 
+/**
+ * Calculates seconds remaining in current batch
+ * Assumes local time is accurate and can be used as source of truth
+ */
+export function getSecondsRemainingInBatch(): number {
+  return BATCH_TIME - (getEpoch() % BATCH_TIME)
+}
+
+export function formatSecondsToMinutes(seconds: number): string {
```

Comment:
Cool but wouldn't `date-fns` have a function for that

Reply:
That's what I though, but as far as I can see, their functions are Date/timestamp based. I would need to add those seconds to now() and compare against it with something like https://date-fns.org/v2.9.0/docs/formatDistanceToNow

Or is there a simpler way?

Reply:
```js
const date = new Date(seconds * 1000)
const minutes = date.getMinutes()
const seconds = date.getSeconds()
```
Is it easier? Not really

As for `date-fn` the best I found
```sh
> d.formatDistance(17000, 0, { includeSeconds: true})
'less than 20 seconds'
> d.formatDistance(97000, 0, { includeSeconds: true})
'2 minutes'
> d.formatDistance(77000, 0, { includeSeconds: true})
'1 minute'
```

which is not ideal

Reply:
Not much better in my opinion. I'd like to keep the current version in this case.